### PR TITLE
Support for raw Data type

### PR DESCRIPTION
**Pull request by Vyeczorny**

### CODE REVIEW

**Review by 3-n — APPROVED** [review]

**Review by mateusznadolski — APPROVED** [review]

File: C0T0/Classes/ApiDataSource/URLSessionApiDataSource.swift [hunk 1]
```diff
@@ -53,14 +53,18 @@ final class URLSessionApiDataSource: ApiDataSource {
...
-                do {
-                    let sanitizedData = data.isEmpty ? "{}".data(using: .utf8)! : data
-                    let object = try jsonDecoder.decode(T.self, from: sanitizedData)
-                    completion(.success(object))
-                } catch let error {
-                    let apiError = ApiError.cannotParseData(error.localizedDescription)
-                    logger.log(apiError)
-                    completion(.failure(apiError))
+                if T.self == Data.self {
+                    completion(.success(data as! T))
```
Comment (TomaszPietrowski) [inline C0T0/Classes/ApiDataSource/URLSessionApiDataSource.swift:57]: Do you need this force casting here? What is the type of data returned by sender?

File: C0T0/Classes/ApiDataSource/URLSessionApiDataSource.swift [hunk 2]
```diff
@@ -53,14 +53,18 @@ final class URLSessionApiDataSource: ApiDataSource {
...
+                } else {
+                    do {
+                        let sanitizedData = data.isEmpty ? "{}".data(using: .utf8)! : data
```
Comment (TomaszPietrowski) [inline C0T0/Classes/ApiDataSource/URLSessionApiDataSource.swift:60]: What's the difference between using `sanitizedData` and passing empty data to the `jsonDecoder`?

File: C0T0/Classes/ApiDataSource/URLSessionApiDataSource.swift [hunk 3]
```diff
@@ -53,14 +53,18 @@ final class URLSessionApiDataSource: ApiDataSource {
...
+                        let object = try jsonDecoder.decode(T.self, from: sanitizedData)
+                        completion(.success(object))
+                    } catch let error {
+                        let apiError = ApiError.cannotParseData(error.localizedDescription)
+                        logger.log(apiError)
```
Comment (TomaszPietrowski) [inline C0T0/Classes/ApiDataSource/URLSessionApiDataSource.swift:65]: 🔍 Side effects 🤓 Don't require changes here, just trolling 😉

**Review by TomaszPietrowski — APPROVED** [review]

File: C0T0/Classes/ApiDataSource/URLSessionApiDataSource.swift (hunk 1 above)
Comment (Vyeczorny) [reply C0T0/Classes/ApiDataSource/URLSessionApiDataSource.swift:57]: Yes, this function doesn't know what the type `T` is, so I need to force cast it here.

File: C0T0/Classes/ApiDataSource/URLSessionApiDataSource.swift (hunk 2 above)
Comment (Vyeczorny) [reply C0T0/Classes/ApiDataSource/URLSessionApiDataSource.swift:60]: Empty data is not a valid json, but in my opinion it should be converted to empty struct, hence this condition.

**Pull request merged**